person.rb

```diff
@@ -1,18 +1,31 @@
-class Person
-  attr_reader :id
+require './Interface/nameable'
+require './rental'
+
+class Person < Nameable
+  attr_reader :id, :rentals
   attr_accessor :name, :age
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
+    super()
   end
 
   def can_use_services?
     of_age? || @parent_permission
   end
 
+  def correct_name
+    @name
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental)
+  end
```

Comment:
- [ ] Please pass the `book` and `date` a person makes a rental. 

See example:
```ruby
  def add_rental(book, date)
    Rental.new(date, book, self)
  end
```

Reply:
- [ ] Please address this

Reply:
- Kindly note that to achieve the Many-to-many association you have to pass the necessary parameters into the `add_rental ` method which you set in your rental class ` def initialize(date, person, book) `  so that when creating a new rental it should belong to a book and also a person and should have a date.
- so in your` book.rb` file the parameter book would be replaced with `self` which enables you to access the instance of the class book.